Python/bltinmodule.c

```diff
@@ -804,25 +804,39 @@ builtin_compile_impl(PyObject *module, PyObject *source, PyObject *filename,
     if (is_ast == -1)
         goto error;
     if (is_ast) {
+        PyArena *arena = _PyArena_New();
```

Comment:
I wonder if anyone has perf-sensitive code that does `ast.parse` on an AST as a no-op, and what the impact on that code will be of allocating and freeing an unused arena. Probably not an issue? I think it would be possible, but slightly more complex, to still avoid arena allocation in the no-op case.